pkg/workflow/controllers/serial_node_reconciler.go

```diff
@@ -52,93 +65,224 @@ func (it *SerialNodeReconciler) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, nil
 	}
 
+	it.logger.V(4).Info("resolve serial node", "node", request)
+
+	err = it.syncChildrenNodes(ctx, node)
```

Comment:
Can you add some comments for `syncChildrenNodes` function? It is difficult for me to understand the logic and meaning of this function. I don't know what to sync and why to sync. 😂